### ISSUE TITLE
Webserver on OSX, can't get IP address.

### ISSUE THREAD

**Issue by Lord-Kamina** [performous/performous#226]
Once again, webserver appears to be functioning as expected, but I noticed this while running on OSX:

![0 2017-02-17 at 01 12 52](https://cloud.githubusercontent.com/assets/6446892/23052225/77d269b6-f4ae-11e6-9655-61538612032a.PNG)

**Comment by Baklap4** [comment]
Currently the webserver is trying to reach http://google.com to try to resolve the ip address: 

```
std::string ScreenIntro::getIPaddr() {
	try {
		boost::asio::io_service netService;
		boost::asio::ip::udp::resolver resolver(netService);
		boost::asio::ip::udp::resolver::query query(boost::asio::ip::udp::v4(), "google.com", ""); //it's a bit of a dirty hack, but it works!
		boost::asio::ip::udp::resolver::iterator endpoints = resolver.resolve(query);
		boost::asio::ip::udp::endpoint ep = *endpoints;
		boost::asio::ip::udp::socket socket(netService);
		socket.connect(ep);
		boost::asio::ip::address addr = socket.local_endpoint().address();
		return addr.to_string();
	} catch(std::exception& e) {
			return "cannot obtain IP";
	}

	return "IP address";
}
```
This is a dirty hack and works only if you have internet. Or a timeout will occur and you will get the message you just got..

**Comment by nieknooijens** [comment]
It's dirty, but most laptops have multiple network adapters. By resolving a
request to Google and read the address from the response i get the ip from
the card that's connected. It's the simplest way to solve this problem.

2017/02/17 8:52 "Arjan Spieard" <notifications@github.com>:

**Comment by Baklap4** [comment]
Yeah so true i don't have a better solution anyway 👍 Maybe instead of showing 'cannot obtain IP' show hostname if available? If hostname is not available, show localhost address if localhost is also not available show the message no devices up?

**Comment by nieknooijens** [comment]
I'll add the "cleanup" tag then since it's technically not a bug, but it's open for improvement

**Comment by Lord-Kamina** [comment]
I find it odd it would time-out, though.
Anyway... An idea I read somewhere that might be useful. Couldn't you
perhaps send a test request to the server and then have the server return
what it sees the IP as?

**Comment by nieknooijens** [comment]
@Lord-Kamina  that would work, but where to send it to? I know google will be always up.

**Comment by Lord-Kamina** [comment]
To the webserver itself. Or would it just return 127.0.0.1?

**Comment by nieknooijens** [comment]
@Lord-Kamina the webserver listens at 127.0.0.1`when set to localhost only, when set to all connections it returns 0.0.0.0

**Comment by Lord-Kamina** [comment]
What I meant was, perhaps set up minimal cases of http client and server
using cpp-netlib itself and have the client ask the server what IP it
thinks it got the request from. Don't know if it'd actually work though.

**Comment by MagicChocoRolls** [comment]
Just an addition to the dirty google hack: Why do u use google.com in the first place? What about v4.ident.me or v6.ident.me? Those sites only return your IP address and nothing more :) The suggestion from Baklap4 seems good to me.


Another question? Can you publish the webserver build you got? ...or is the code already pushed to the repo and I can build it myself using correct dependencies and so on?

**Comment by nieknooijens** [comment]
Didn't really think about it TBH. I googled for hours to find an "offline"
way, until I came up with "resolve a request and check the return address"
then I needed something to send my request to, Knowing that it would always
be up and always respond. So I just decided on Google because if you got
internet access, you got Google. (Unless you live in china)

2017/03/02 15:55 "MagicChocoRolls" <notifications@github.com>:

**Comment by Baklap4** [comment]
@MagicChocoRolls the code for the webserver has been in master for quite some time, just build it and install ;)

**Comment by MagicChocoRolls** [comment]
@Baklap4 Tried that a few month ago by following the Wiki, but that resulted in a broken build environment for me. I am not the every day Mac-Compile-Guy, so I have to read through everything and try to understand what is needed and where I get it.
Will try to checkout the code and build it with clang now :)

**Comment by Baklap4** [comment]
@MagicChocoRolls you can join our [discord ](https://discord.gg/NS3m3ad)or [irc channel](http://webchat.freenode.net/?channels=#performous) to get support for this ;)
@Lord-Kamina is the one building at a mac if i'm correctly maybe he can help you set it up?

**Comment by Baklap4** [comment]
Closed since #285 is merged
Opened #286 to fix the shown text
